From e8cfcd9705d5a59c3f662ec52de063478b1774ed Mon Sep 17 00:00:00 2001
From: Tristan Cacqueray <tdecacqu@redhat.com>
Date: Sun, 31 Jan 2016 11:15:36 -0500
Subject: [PATCH] Add poll.poll() timeout

This patch uses poll timeout to avoid process being stuck when there
is no connection activity. Otherwise, when the gerrit server is quiet,
poll() may wait infinitely preventing stop or reload to finish.

Before the 'connection' concept, signals were properly interrupting the
poll system call and would exit gracefully.

Change-Id: If3129a65da6119acf69ad00e2e78c7ec82a49941
Closes-Bug: https://storyboard.openstack.org/#!/story/2000472
---

diff --git a/tests/base.py b/tests/base.py
index f3bfa4e..9ac28de 100755
--- a/tests/base.py
+++ b/tests/base.py
@@ -510,7 +510,7 @@
             poll = select.poll()
             poll.register(self.sock, select.POLLIN)
             poll.register(self.wake_read, select.POLLIN)
-            ret = poll.poll()
+            ret = poll.poll(500)
             for (fd, event) in ret:
                 if fd == self.sock.fileno():
                     data = self.sock.recvfrom(1024)
diff --git a/zuul/connection/gerrit.py b/zuul/connection/gerrit.py
index f8e5add..82734da 100644
--- a/zuul/connection/gerrit.py
+++ b/zuul/connection/gerrit.py
@@ -154,7 +154,7 @@
         poll = select.poll()
         poll.register(stdout.channel)
         while not self._stopped:
-            ret = poll.poll()
+            ret = poll.poll(500)
             for (fd, event) in ret:
                 if fd == stdout.channel.fileno():
                     if event == select.POLLIN:
